ui/desktop/src/components/ChatInput.tsx

```diff
@@ -143,6 +144,27 @@ export default function ChatInput({
   const [showCreateRecipeModal, setShowCreateRecipeModal] = useState(false);
   const [showEditRecipeModal, setShowEditRecipeModal] = useState(false);
   const [isFilePickerOpen, setIsFilePickerOpen] = useState(false);
+  const [sessionWorkingDir, setSessionWorkingDir] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (!sessionId) {
+      setSessionWorkingDir(null);
+      return;
+    }
+
+    const fetchSessionWorkingDir = async () => {
+      try {
+        const response = await getSession({ path: { session_id: sessionId } });
+        if (response.data?.working_dir) {
+          setSessionWorkingDir(response.data.working_dir);
+        }
+      } catch (error) {
+        console.error('[ChatInput] Failed to fetch session working dir:', error);
+      }
+    };
```

Comment:
if we fetch the working dir here, shouldn't we just pass that on to the DirSwitcher instead of having it pick its own? probably be nicer if the dirswitcher had a callback on what should happen when the dir switched